src/txn/txn_rollback_to_stable.c

```diff
@@ -72,6 +72,10 @@ __rollback_abort_update(WT_SESSION_IMPL *session, WT_UPDATE *first_upd,
      */
     if (first_upd != NULL) {
         F_CLR(first_upd, WT_UPDATE_HS);
+
+        /* Clear the history store flag of an update following a stable tombstone. */
+        if (first_upd->type == WT_UPDATE_TOMBSTONE && first_upd->next != NULL)
```

Comment:
Shall we walk through all the aborted tx in the middle? Though there shouldn't be any. May be we can assert it must be a full update restored from hs here.

Reply:
I am also curious in knowing if we are always certain that the next update (after tombstone) will never be an aborted update.

Reply:
It is possible that there can be aborted updates in between. Added additional checks to consider this scenario. The existing test is enhanced to consider this scenario.